bugbug/models/accessibility.py

```diff
@@ -0,0 +1,157 @@
+                            ("title", self.text_vectorizer(min_df=0.0001), "title"),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.0001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+                ("sampler", BorderlineSMOTE(random_state=0)),
+                (
+                    "estimator",
+                    xgboost.XGBClassifier(n_jobs=utils.get_physical_cpu_count()),
+                ),
+            ]
+        )
+
+    @staticmethod
+    def __is_accessibility_bug(bug):
+        """Check if a bug is an accessibility bug."""
+        return bug["cf_accessibility_severity"] != "---" or "access" in bug["keywords"]
+
+    @staticmethod
+    def __download_older_access_bugs():
+        """Retrieve accessibility related bugs newer than 4 years and 6 months ago.
+
+        By including older accessibility bugs, this function extends the dataset used
+        for model training compared to the default, which only considers bugs from 2 years
+        and 6 months ago. This extension in the time frame aims to improve the performance
+        of the model by providing a more comprehensive set of historical data.
+        """
+        lookup_start_date = datetime.utcnow() - relativedelta()
+        params = {
+            "f1": "creation_ts",
+            "o1": "greaterthan",
+            "v1": lookup_start_date.strftime("%Y-%m-%d"),
+            "f2": "OP",
+            "j2": "OR",
+            "f3": "cf_accessibility_severity",
+            "o3": "notequals",
+            "v3": "---",
+            "f4": "keywords",
+            "o4": "substring",
+            "v4": "access",
+            "f5": "CP",
+            "product": bugzilla.PRODUCTS,
+        }
+
+        older_access_bugs_ids = bugzilla.get_ids(params)
+        bugzilla.download_bugs(older_access_bugs_ids)
+
+    def get_labels(self):
+        classes = {}
+
+        logger.info("Downloading older accessibility bugs...")
+        self.__download_older_access_bugs()
+
+        for bug in bugzilla.get_bugs():
+            bug_id = int(bug["id"])
+
+            if "cf_accessibility_severity" not in bug:
+                continue
+
+            classes[bug_id] = 1 if self.__is_accessibility_bug(bug) else 0
+
+        positive_samples = sum(label == 1 for label in classes.values())
+        negative_samples = sum(label == 0 for label in classes.values())
+
+        logger.info(
+            "%d bugs are classified as non-accessibility",
+            negative_samples,
+        )
+        logger.info(
+            "%d bugs are classified as accessibility",
+            positive_samples,
+        )
+
+        ratio = round((negative_samples / positive_samples) ** 0.5)
+
+        self.clf.named_steps["estimator"].set_params(
+            scale_pos_weight=ratio, subsample=0.5
+        )
```

Comment:
Having this here seems tricky! It would be better to do any tuning in the constructor or the training method. Also, we are using the testing set to configure the model, which should be avoided.

Reply:
Im working on an experiments testing the effect of working with or without changing the hyperparemeters this way.